src/shared/components/palette/base-field/abstract-form-field.component.ts

```diff
@@ -0,0 +1,34 @@
+import { Input } from '@angular/core';
+import { CaseField } from '../../../domain/definition';
+import { AbstractControl, FormControl, FormGroup } from '@angular/forms';
+
+export abstract class AbstractFormFieldComponent {
+
+  @Input()
+  caseField: CaseField;
+
+  @Input()
+  formGroup?: FormGroup;
+
+  @Input()
+  registerControl?: <T extends AbstractControl> (control: T) => T;
+
+  protected defaultControlRegister(): (control: FormControl) => AbstractControl {
+    return control => {
+      if (!this.formGroup) {
+        return null;
+      }
+      if (this.formGroup.controls[this.caseField.id]) {
```

Comment:
Will the `this.caseField` always be there?

Reply:
I abstracted out this class from AbstractFieldWriteComponent and AbstractFieldReadComponent, both had casefield mandatory